### DISCUSSION TITLE
DataDriven(Sheets) Testing

### DISCUSSION THREAD

**Discussion by abdurraheemj** [seleniumbase/SeleniumBase#1035] [Q&A] [answered]
Hi Michael,
Do you have example script to read data from google sheet? would like to assert page content. feeding data into sheets and write script to read and assert

**Comment by mdmintz** [comment, marked as answer]
@abdurraheemj You can use ``self.get_text(selector)`` to scrape data from the sheet. You can use ``self.assert_text(text, selector)`` to assert data directly from the sheet. You can use ``self.type(selector, text)`` to type data into a sheet. I don't have an existing example that uses Google Sheets, but the standard commands should work.

**Reply by abdurraheemj** [reply]
Challenging
